packages/zowe-explorer/src/utils/SettingsConfig.ts

```diff
@@ -67,15 +67,51 @@ export class SettingsConfig {
         );
     }
 
+    /**
+     * Checks if the Zowe Explorer major version does not match
+     * the version defined in the user's `settings.json` file.
+     *
+     * @param userVersion The user's version defined in settings.json
+     * @param versionString The current version string for Zowe Explorer
+     * @returns false if the version *roughly* matches, true otherwise - e.g.:
+     *
+     * `majorVersionMismatch("2.6.1", "2")` will return `false`
+     *
+     * `majorVersionMismatch("3", "2")` will return `true`
+     */
+    public static majorVersionMismatch(userVersion: string, versionString: string): boolean {
+        if (userVersion == null) {
+            return true;
+        }
+
+        if (userVersion.startsWith(versionString)) {
+            return false;
+        }
+
+        return userVersion !== versionString;
+    }
+
     public static async standardizeSettings(): Promise<void> {
         ZoweLogger.trace("SettingsConfig.standardizeSettings called.");
-        const globalIsNotMigrated =
-            SettingsConfig.configurations.inspect(globals.SETTINGS_VERSION).globalValue !== SettingsConfig.currentVersionNumber;
-        const workspaceIsNotMigrated =
-            SettingsConfig.configurations.inspect(globals.SETTINGS_VERSION).workspaceValue !== SettingsConfig.currentVersionNumber;
+        // Need to coerce all possible version values to a string to correct previous values
+        const globalVersion = String(SettingsConfig.configurations.inspect(globals.SETTINGS_VERSION).globalValue);
+        const workspaceVersion = String(SettingsConfig.configurations.inspect(globals.SETTINGS_VERSION).workspaceValue);
+        const currentVersion = String(SettingsConfig.currentVersionNumber);
+
+        const globalIsNotMigrated = SettingsConfig.majorVersionMismatch(globalVersion, currentVersion);
+        const workspaceIsNotMigrated = SettingsConfig.majorVersionMismatch(workspaceVersion, currentVersion);
         const workspaceIsOpen = vscode.workspace.workspaceFolders !== undefined;
         const zoweSettingsExist = SettingsConfig.zoweOldConfigurations.length > 0;
 
+        // Update version value if there is a mismatch (coerces previous values into numeric ["2.6.1" -> 2])
+        if (globalVersion == null || globalVersion !== currentVersion) {
+            SettingsConfig.setDirectValue(globals.SETTINGS_VERSION, SettingsConfig.currentVersionNumber);
+        }
+        if (workspaceVersion != null && workspaceVersion !== currentVersion) {
+            SettingsConfig.setDirectValue(globals.SETTINGS_VERSION, SettingsConfig.currentVersionNumber, vscode.ConfigurationTarget.Workspace);
+        }
```

Comment:
I'm not sure I agree with forcing the `zowe.settings.version` to match the ZE major version.
I can understand why, but it's possible that we may need to refactor a few things within the same major release.
Not saying we would do that this (or the next) release.

On the contrary, what if ZE settings are stable enough where the settings stay the same even after a new major release or ZE?

I guess I'm comparing this to the CLI zowe.schema.json where that is still at version `"1.0"`.

Feel free to ignore this if it sounds crazy. 😋

Reply:
I agree with @zFernand0 that if the next release 3.x does not require any migration that the value should stay 2 until there is a change, because changing this value is already a migration that can have a major impact (e.g. if the user has dozens of Git repos with workspace settings that will change requiring a commit.)

Supporting migration if there is not major version number update probably needs more discussion with use cases. In general breaking changes that require migration (even if semi-automatic) should be a major version update with the same argument as above as people keep their workspace settings in Git and require updates.

Reply:
@phaumer This logic won't exist in `next` (v3), as once we merge the PR for local storage #2281 we won't need this setting anymore. For now, this fix should help coerce any version strings that don't match to be updated to the integer `2`, and once `main` is updated with changes from `next` I will remove it 👍